packages/asl-projects/client/pages/sections/protocols/steps.js

```diff
@@ -449,7 +449,7 @@ const StepsRepeater = ({ values, prefix, updateItem, editable, project, isReview
     onSave={steps => {
       // Create deep clones of reusable steps to ensure immutability
       const reusableSteps = steps
-        .filter(step => step.reusable && (step.completed || step.saved))
+        .filter(step => step.reusable && step.existingValues)
```

Comment:
I worry that this going to affect behaviour around how steps are persisted. From the names this is filtering out steps where the user hasn't hit save from being persisted to the list of reusable steps, so could cause draft steps to start showing as an option earlier than the user expects.

Is there an equivalent fix that excludes uncompleted/saved steps from the clientside comparison data rather than changing the behaviour of saving steps?